File: Chapitre2/exercice2.4.py
import os
import sys

i = 0

def main():
    global i
    for i in range(1, 6):
        fils = os.fork()

        if fils == -1:
            print("Erreur lors de la création d'un des processus", file=sys.stderr)
            sys.exit(1)

        if fils == 0:  # I am in the child
            print(f"Je suis dans le fils{i}: moi={os.getpid()},père={os.getppid()}")
            break

if __name__ == "__main__":
    main()

File: Chapitre2/exercice2.4.c
//créer 5 processus fils avec for et break sans synchronisation
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <errno.h>

int i;

int main()
{
    pid_t fils;
    for(i=1; i<=5 ; i++)
    {
        fils=fork();

        if( fils == -1)
        {
            perror("Erreur lors de la création d'un des processus");
            exit(1);
        }
        if(fils == 0) //je suis dans le fils
        {
            printf("Je suis dans le fils%d: moi=%d,père=%d\n",i,getpid(),getppid());
            break;
        }
        // if(fils > 0)  //dans le père
        // {
        //     printf("Je suis dans le père: fils%d =%d, moi=%d, mon_père=%d\n",i, fils , getpid(), getppid());
        // }
    }
    
    return 0;
}